src/services/WebSocketService.ts

```diff
@@ -20,12 +20,27 @@ export class WebSocketService {
       ws.on('message', (message: string) => {
         const data = JSON.parse(message);
         if (data.type === 'toolResponse') {
-          const toolResponses = JSON.parse(data.response); // Consider checking if you can send JSON directly to avoid parsing.
-          toolResponses.forEach((toolResponse: { function_name: string; response: string }) => {
+          let toolResponse = data.response;
```

Comment:
This is the only important bit right? the rest of the changes need to go before merging, they are already in another PR that we already need to change, we don't wanna make the changes twice